### ISSUE TITLE
Remove LearnLogger?

### ISSUE THREAD

**Issue by ericphanson** [beacon-biosignals/Lighthouse.jl#61]
After #60 we've defined an interface that folks can plug into for custom logging.

We could drop the Tensorboard dep by removing the `LearnLogger`. We could make a no-op logger that doesn't actually log anything, or maybe a `Dict` logger that logs things to a `logged` dict as the current one does, which could be good for testing. Or we could keep the current LearnLogger.

**Comment by hannahilea** [comment]
> maybe a Dict logger that logs things to a logged dict as the current one does, which could be good for testing

I vote this one!

And then either drop [tb] `LearnLogger` entirely, or perhaps add it to the docs or tests as a (tested?) example? It feels bad to have to keep supporting it, though, since as far as we know no one uses it. Maybe dropping it is fine, esp when our current internal logger is public (soon?) and we can point at it as an example of how to implement the interface.

**Comment by ericphanson** [comment]
One other option could be to have `LearnLogger` wrap other loggers. So it would be like
```julia
struct LearnLogger{L}
    logger::L
    logged::Dict{String,Vector{Any}}
end
```
and it could log everything to `logged` and then forward the call onto `logger`... So like
```julia
function log_value!(logger::LearnLogger, field, value)
    values = get!(() -> Any[], logger.logged, field)
    push!(values, value)
    return log_value!(logger.logger, field, value)
end
```
Then a `LearnLogger` is just something that wraps a logger to store the info to a `Dict` as well. Not really sure if we would actually use that or not though. I think the `Dict`-based thing is kinda problematic for huge training jobs because you need to store everything in memory.

**Comment by hannahilea** [comment]
> I think the Dict-based thing is kinda problematic for huge training jobs because you need to store everything in memory.

Yeah---and I think if someone _does_ want to log to a local dict, they can easily make a logger that does that.
